project/cmake/weblib/gen.txt

```diff
@@ -0,0 +1,27 @@
+set(canvas_h_generator ${CMAKE_CURRENT_SOURCE_DIR}/${root}/tools/ci/type/gen-canvas-h.cjs)
+set(canvas_js_generator ${CMAKE_CURRENT_SOURCE_DIR}/${root}/tools/ci/type/gen-canvas-js.cjs)
+set(canvas_yaml ${appsDir}/weblib/canvas.yaml)
+set(canvas_h ${appsDir}/weblib/canvas.h)
+set(canvas_js ${appsDir}/weblib/canvas.js)
+
+add_custom_command(
+    OUTPUT ${canvas_h}
+    COMMAND node ${canvas_h_generator}
+    DEPENDS ${canvas_h_generator} ${canvas_yaml}
+    COMMENT "Generating canvas.h file..."
+    VERBATIM
+)
+
+add_custom_command(
+    OUTPUT ${canvas_js}
+    COMMAND node ${canvas_js_generator}
+    DEPENDS ${canvas_js_generator} ${canvas_yaml}
+    COMMENT "Generating canvas.js file..."
+    VERBATIM
+)
```

Comment:
Yes. That is the case. That part will be done by @veghdev

Reply:
On secound though, I don't see the reason why do we build the web interface in the desktop build.

Reply:
We cannot run executable unit tests on the weblib when it's not compilable with desktop build.

Reply:
Yes we could, unit tests can be run with node. However there are no unit tests for weblib.

Reply:
It's not an executable unit test, because it need another tool `node` to run the code.
I wanted to make the Qt code compatible with this c-api to work exactly the same as run in node.

Reply:
We should discuss that plan separately. The dev.env/cmake config can be changed again in the future when needed. For now this is adequate for the current implementation/needs.